account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -365,3 +365,101 @@ const overrideField = <
   }
   return undefined;
 };
+
+/**
+ * Utility function to generate a signed Permit for erc20 transaction
+ *
+ * @param {UserOperationRequest<TEntryPointVersion>} userOp - The user operation request
+ * @param {AlchemySmartAccountClient} client - The Alchemy smart account client
+ * @param {TAccount} account - The smart account instance
+ * @param {string | string[]} policyId - The policy ID or array of policy IDs
+ * @param {PolicyToken} policyToken - The policy token configuration
+ * @param {Address} policyToken.address - ERC20 contract address
+ * @param {bigint} [policyToken.maxTokenAmount] - Optional ERC20 token limit
+ * @param {"NONE" | "PERMIT"} [policyToken.approvalMode] - ERC20 approve mode
+ * @param {string} [policyToken.erc20Name] - EIP2612 specified ERC20 contract name
+ * @param {string} [policyToken.version] - EIP2612 specified ERC20 contract version
+ * @returns {Promise<Hex>} Returns a Promise containing the signed EIP2612 permit
+ */
+const generateSignedPermit = async <
+  TAccount extends SmartContractAccount,
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
+>(
+  userOp: UserOperationRequest<TEntryPointVersion>,
+  client: AlchemySmartAccountClient,
+  account: TAccount,
+  policyId: string | string[],
+  policyToken: {
+    address: Address;
+    maxTokenAmount?: bigint;
+    approvalMode?: "NONE" | "PERMIT";
+    erc20Name?: string;
+    version?: string;
+  }
+): Promise<Hex> => {
+  if (!client.chain) {
+    throw new ChainNotFoundError();
+  }
+  if (!policyToken.erc20Name || !policyToken.version) {
+    throw new Error("erc20Name or version is missing");
+  }
+  // get a paymaster address
+  const maxAmountToken = policyToken.maxTokenAmount || maxUint256;
+  const paymasterData = await (client as AlchemySmartAccountClient).request({
```

Comment:
this client is already typed to this above
```suggestion
  const paymasterData = await client.request({
```